### ISSUE TITLE
Training on higher resolution on custom dataset

### ISSUE THREAD

**Issue by kHarshit** [cogaplex-bts/bts#49]
Hi, 

I want to train bts (pytorch version) on high resolution images (around (960, 2500)). How do I modify the network, if needed? Is it possible considering the pretrained encoder is used?

Right now, I'm getting the error in the model:

```
Traceback (most recent call last):
  File "bts_main.py", line 611, in <module>
    main()
  File "bts_main.py", line 607, in main
    main_worker(args.gpu, ngpus_per_node, args)
  File "bts_main.py", line 451, in main_worker
    lpg8x8, lpg4x4, lpg2x2, reduc1x1, depth_est = model(image, focal)
  File "/nfs/interns/kharshit/miniconda3/envs/pylatest/lib/python3.7/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/nfs/interns/kharshit/miniconda3/envs/pylatest/lib/python3.7/site-packages/torch/nn/parallel/data_parallel.py", line 150, in forward
    return self.module(*inputs[0], **kwargs[0])
  File "/nfs/interns/kharshit/miniconda3/envs/pylatest/lib/python3.7/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/nfs/interns/kharshit/DepthEstimation/bts/pytorch/bts.py", line 322, in forward
    return self.decoder(skip_feat, focal)
  File "/nfs/interns/kharshit/miniconda3/envs/pylatest/lib/python3.7/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/nfs/interns/kharshit/DepthEstimation/bts/pytorch/bts.py", line 201, in forward
    concat5 = torch.cat([upconv5, skip3], dim=1)
RuntimeError: invalid argument 0: Sizes of tensors must match except in dimension 1. Got 125 and 124 in dimension 3 at /opt/conda/conda-bld/pytorch_1565272271120/work/aten/src/THC/generic/THCTensorMath.cu:71
```

Thanks!

**Comment by cogaplex-bts** [comment]
Due to the strided convolutions and the simple upconv layers in our network, the height and width of the input should be multiples of 32.

**Comment by kHarshit** [comment]
Thanks, I'll try training by making image size to be multiple of 32.

**Comment by kHarshit** [comment]
Hi @cogaplex-bts,

It works really well. I also want to know how do I apply Transfer learning in order to train bts on my custom dataset.

I tried training by using the pretrained encoder using NYU pretrained weights, instead of using ImageNet pretrained weights (and training the decoder from scratch). I did the following changes:

In `bts.py`, to load pretrained NYU V2 weights of `densenet121` encoder (here `BtsModelOriginal` is the original `bts` model required to load weights): 

https://github.com/cogaplex-bts/bts/blob/e63ccc00a1d44cc417c745714e3cabe58559f054/pytorch/bts.py#L273-#L276

```   
     if True:
             model = BtsModelOriginal(params)
             model = torch.nn.DataParallel(model)
             checkpoint = torch.load('/kharshit/DepthEstimation/bts/models/bts_nyu_v2_pytorch_densenet121/model')
             model.load_state_dict(checkpoint['model'])
             self.base_model = model.module.encoder.base_model
             self.feat_names = ['relu0', 'pool0', 'transition1', 'transition2', 'norm5']
```

And, everything the same. But, the output contains lines in the depth images as shown. 

8x8 color cmap output: (similar for 4x4, 2x2):
![kona_dataset_14_003_color_8x8](https://user-images.githubusercontent.com/20869671/75956360-bc68f580-5edd-11ea-99a7-e700ab894c94.png)

I think I need to modify some layers as well, but I'm not sure which ones.
